metrics/eval_video.py

```diff
@@ -0,0 +1,168 @@
+from utils.video_info import VideoInfo
+from utils.video_eval_method import VideoEvalMethodVmaf, VideoEvalMethod
+from utils.video_align_method import VideoAlignMethodFfmpeg, VideoAlignMethod, VideoAlignMethodOcr
+from eval_audio import get_remote_ground
+
+
+description = \
+'''
+This script provide multi methods to evaluate network quality.
+For example, the method of Vmaf https://github.com/Netflix/vmaf.
+'''
+
+class VideoEvaluation(object):
+    def __init__(self, eval_method : VideoEvalMethod, align_method : VideoAlignMethod, args):
+        self.eval_method = eval_method
+        self.align_method = align_method
+        self.args = args
+
+    ### ffmpeg orders below ###
+
+    def change_video_type(self, video_info : VideoInfo, new_type):
+        output = NamedTemporaryFile('w+t', suffix=".%s" % (new_type))
+        cmd = ["ffmpeg"]
+        if video_info.format_name == "rawvideo":
+            cmd.extend(["-video_size", video_info.video_size])
+        cmd.extend(["-r", defalut_fps])
+        cmd.extend(["-i", video_info.video_path, "-y"])
+        if new_type == "yuv":
+            cmd.extend(["-video_size", video_info.video_size])
+        cmd.extend(["-r", defalut_fps])
+        cmd.append(output.name)
+        subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding="utf8")
+
+        return output
+
+    def eval(self, src_video_path, dst_video_path):
+        # get the correspond VideoInfo according to path
+        src_video_info = VideoInfo(src_video_path, self.args.video_size)
+        dst_video_info = VideoInfo(dst_video_path, self.args.video_size)
+
+        print("src_video_info:", src_video_info.__dict__)
+        print("dst_video_info:", dst_video_info.__dict__)
+        
+        # init video size
+        if not src_video_info.video_size and not dst_video_info.video_size:
+            raise ValueError("Please specify the arguments --video_size")
+ 
+        if not dst_video_info.video_size:
+            dst_video_info.update_video_size(src_video_info.video_size)
+        elif not src_video_info.video_size:
+            src_video_info.update_video_size(dst_video_info.video_size)
+
+        # update video info
+        src_video_info.parse_video_by_ffprobe(video_size=src_video_info.video_size)
+        dst_video_info.parse_video_by_ffprobe(video_size=dst_video_info.video_size)
+        video_size = src_video_info.video_size
+
+        # check video type
+        if src_video_info.format_name not in self.eval_method.support_type:
+            fo_new_src_video = self.change_video_type(src_video_info, self.eval_method.support_type_abbreviation[0])
+            src_video_info = VideoInfo(fo_new_src_video.name)
+        if dst_video_info.format_name not in self.eval_method.support_type:
+            fo_new_dst_video = self.change_video_type(dst_video_info, self.eval_method.support_type_abbreviation[0])
+            dst_video_info = VideoInfo(fo_new_dst_video.name)
```

Comment:
since the only difference of these two blocks are source video and destination video, so I guess you can create a function and pass src/dst video as its parameter. You can keep this for now but please rethink the code to see it we can make it better. :-)